pkg/asset-manager-utils/contracts/SinglePoolAssetManager.sol

```diff
@@ -0,0 +1,250 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import "@balancer-labs/v2-vault/contracts/interfaces/IVault.sol";
+
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/IERC20.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/math/Math.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/SafeCast.sol";
+
+import "./IAssetManager.sol";
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+// solhint-disable private-vars-leading-underscore
+abstract contract SinglePoolAssetManager is IAssetManager {
+    using Math for uint256;
+    using SafeCast for uint256;
+
+    uint64 private constant ONE = 1e18;
+
+    /// @notice The Balancer Vault contract
+    IVault public immutable vault;
+
+    /// @notice The id of the pool which owns this asset manager
+    bytes32 public poolId;
+
+    /// @notice The token which this asset manager is investing
+    IERC20 public immutable token;
+
+    /// @notice the total AUM of tokens that the asset manager is aware it has earned
+    uint256 public totalAUM;
+
+    PoolConfig private _poolConfig;
+
+    constructor(
+        IVault _vault,
+        bytes32 _poolId,
+        address _token
+    ) {
+        IERC20(_token).approve(address(_vault), type(uint256).max);
+        vault = _vault;
+        poolId = _poolId;
+        token = IERC20(_token);
+    }
+
+    modifier onlyPoolController() {
+        address poolAddress = address((uint256(poolId) >> (12 * 8)) & (2**(20 * 8) - 1));
+        require(msg.sender == poolAddress, "Only callable by pool controller");
+        _;
+    }
+
+    modifier correctPool(bytes32 pId) {
+        require(pId == poolId, "SinglePoolAssetManager called with incorrect poolId");
+        _;
+    }
+
+    // Investment configuration
+    function _getTargetInvestment(
+        uint256 cash,
+        uint256 managed,
+        uint256 investablePercent
+    ) private pure returns (uint256) {
+        return (cash + managed).mul(investablePercent).divDown(1e18);
```

Comment:
This should be `FixedPoint.mulDown()` (or just `mulDown` if we drop `Math` and use `FixedPoint`)